test-two/src/helpers/http.js

```diff
@@ -1,18 +1,55 @@
 import fetch from "node-fetch";
 
-export const get = async url => {
+export const request = async (url, requestInit = {}) => {
   const response = await fetch(url, {
-    method: "GET",
-    headers: {
-      "User-Agent": "request"
-    }
+    ...requestInit,
   });
 
   const json = await response.json();
 
   if (response.status === 200) {
     return json;
   } else {
-    throw Error(json.Message);
+    const error = json.message || json.Message || json;
+    throw Error(error);
   }
+}
+
+export const get = async (url, requestInit = {}) => {
+  return await request(url, {
+    ...requestInit,
+    method: "GET"
+  });
 };
+
+export const post = async (url, requestInit = {}) => {
+  return await request(url, {
+    ...requestInit,
+    method: "POST"
+  });
+}
+
+export const joinPaginatedRequests = async (
+  total,
+  pageSize,
+  getter,
+) => {
+  let i = 0;
+  let maxPages = Math.ceil(total / pageSize);
+  const requests = [];
+
+  for (let index = 0; index < maxPages; index++) {
+    requests.push(
+      getter(pageSize, index)
+    );
+  }
+
+  const results = await Promise.all(requests);
```

Comment:
Is there a risk here that too many concurrent requests could cause a problem?

Reply:
That is an interesting question.
I thought at first to come up with a naive queue and using worker threads for batching the application, but then I remembered the way the event-loop behaves, and how the different `requests` are stored.
But in a nutshell, I am moving the complexity to the OS to handle the socket access, threading and queuing. 
The OS usually have a pretty good concurrency control over the socket, and I assumed that the load is fairly small.
If the load was maybe in the order of millions, a different approach would be necessary, as the just result object list would cause a out of memory error.